### ISSUE TITLE
Enable or disable KeepingYouAwake on specific (user defined) battery levels

### ISSUE THREAD

**Issue by timbru31** [newmarcel/KeepingYouAwake#24]
Hi,

I've got a suggestion.
Maybe a feature can be added that disables KeepingYouAwake when e.g. the battery level is <10% and re-enables it on either AC connection or when the battery level is >10% again.

Cheers,
Tim

**Comment by derhuerst** [comment]
Maybe this is possible with Automator?

**Comment by KyleKing** [comment]
Good idea! It might be trickier than it sounds to implement in the app, so building your own tool might be the easiest. I found a few ideas:

- I think your best bet is to use a variation of ```pmset -g everything | grep Cycles``` and extract the battery percent to be checked every so often. [This guide](http://applehelpwriter.com/2014/08/25/applescript-make-your-own-battery-health-meter/comment-page-1/) walks you through it.

- The easiest (i.e expensive) solution is the [DssW Power Manager](http://www.dssw.co.uk/powermanager/) and [this guide](http://www.dssw.co.uk/powermanager/scheduleassistant/run-an-automator-workflow-on-battery-remaining.html) would guide you through the whole setup.

- I looked into [Battery-Time-Remaining](https://github.com/codler/Battery-Time-Remaining) (but it only creates a notification and doesn't seem to trigger a script) and [Griffin proxi](http://www.cnet.com/how-to/how-to-trigger-os-x-system-events-with-griffin-proxi/) which doesn't run on my computer

Anyone else know of any better options?

**Comment by newmarcel** [comment]
This might actually be possible with the `IOPowerSources` APIs… I'll look into it.

(Stack Overflow article for reference: http://stackoverflow.com/a/272774 )

**Comment by newmarcel** [comment]
You can now test this feature in the first beta for 1.4.0: https://github.com/newmarcel/KeepingYouAwake/releases/tag/1.4beta1

**Comment by timbru31** [comment]
This is great news! Going to test it :rocket:

**Comment by timbru31** [comment]
The feature works great, I only have two things:

- When KeepingYouAwake you is enabled currently and I adjust the slider it disabled KeepingYouAwake, even when the configured level is not reached
- When the configured level is reached again, KeepingYouAwake could be auto-enabled again.

Thanks,
Tim

**Comment by alexandreleroux** [comment]
Should this issue should be closed, now that it's part of release 1.4.0? Not sure if it has been implemented like @timbru31 suggested though.

**Comment by newmarcel** [comment]
I'll keep this issue open until I have implemented the mentioned feedback.

1) To avoid some internal state handling I decided to deactivate the app when the checkbox or slider for this feature are modified. I'll try to come up with a cleaner solution.

2) This is a tricky one _(auto-enabling the app when you charge your Mac)_ and would require more internal state handling. I'll think about it, but it's likely at the moment, that I won't implement it.
